core/commands/storage.go

```diff
@@ -415,24 +450,27 @@ signature back to the host to complete payment.`,
 			return fmt.Errorf("storage client api not enabled")
 		}
 
-		// verify challenge
 		ssID := req.Arguments[0]
 		challengeHash := req.Arguments[1]
 		chunkHash := req.Arguments[2]
-		if ss := storage.SessionMap[ssID]; ss == nil {
+		// get info from session
+		ss := storage.SessionMap[ssID]
+		if ss == nil {
 			return fmt.Errorf("session id doesn't exist")
 		} else {
+			// time out check
 			now := time.Now()
 			if now.After(ss.Time.Add(challengeTimeOut)) {
```

Comment:
(now is a good time to fix this): time should be per chunk based

Reply:
good catch!!